src/utils.jl

```diff
@@ -73,3 +73,23 @@ function testfeaturecollection(fc)
     @assert geometrycolumns(fc) isa NTuple "feature collection $fc doesn't return a `NTuple` from `geometrycolumns`."
     return true
 end
+
+"""
+    testraster(raster)
+
+Test whether the required interface for your `raster` has been implemented correctly.
+"""
+function testraster(raster)
+    @assert israster(typeof(raster)) "$raster doesn't implement `israster`."
+    @assert trait(raster) isa AbstractRasterTrait "$raster does not return an `AbstractRasterTrait` for `trait`."
+    @assert isnothing(geomtrait(raster))
+    @assert !isnothing(crs(raster)) "Raster $raster doesn't return anything for `crs`"
+    l, t = affine(raster)
+    @assert l isa AbstractMatrix{<:Real} "Raster $raster doesn't return an `AbstractMatrix{<:Real}` for `affine`"
+    @assert t isa AbstractVector{<:Real} "Raster $raster doesn't return an `AbstractVector{<:Real}` for `affine`"
+    @assert size(l)[1] == size(l)[2] "Raster $raster doesn't return a square matrix for `affine`"
+    @assert size(l)[1] == Base.length(t) "Raster $raster doesn't return the same dimensions for the linear and translation part of `affine`"
```

Comment:
Should we allow `nothing` as a return value form `affine` For netcdf?

Reply:
Hmm yes. And in the future we should have a method that specifies the coordinate mapping? If that returns Affine, affine should not return nothing?